### ISSUE TITLE
Column name tab completion in REPL

### ISSUE THREAD

**Issue by anandijain** [JuliaData/DataFrames.jl#1965]
I don't know enough about julia to know how easy this would be, but it would be an amazing feature if you could tab complete column names.

Example:
Say df has columns "first", "second", "third"

julia> `df.f ` *TAB* -> `df.first`
julia> `groupby(df, :f ` *TAB* -> `groupby(df, :first)`

What do people think?
Thanks

**Comment by bkamins** [comment]
Tab completion of `df.f` to `df.first` works in REPL.

I think the second example (with `groupby` is impossible to do generically).

**Comment by nalimilan** [comment]
Yes, with `groupby`, `select` and friends, we would need a mechanism that the REPL/IDEs could use to get a list of possible values for remaining arguments, taking into account the type of the first argument. That would be very convenient. You can file an issue against Julia and/or your IDE to discuss that (and link to this issue).

**Comment by bkamins** [comment]
@nalimilan for `groupby` etc. to work it would require the REPL/IDE to conditionally depend on DataFrames.jl specifically, which I think is not something that would be acceptable in general (that is why I have closed the issue) - but maybe I am wrong here.

What I mean is that it is fully possible, that some other package defines `groupby` etc. function with a completely different call pattern and meaning of columns (of course it is probably not a best practice to do this, but it is a fully viable thing). Then IDE/REPL would try to use the rules from DataFrames.jl and apply it to some other package.

**Comment by bkamins** [comment]
After thinking about it another option could be to define in DataFrames.jl functionalities for REPL/IDE that would be conditionally loaded when DataFrames.jl is loaded. This is probably doable if someone would be willing to implement it, so I reopen the issue.

**Comment by nalimilan** [comment]
I was thinking about having the REPL define a function like `complete_nextargs` that it would call like this: `complete_nextargs(f, firstargs)`. DataFrames would add methods to that function retuning possible values for the next argument, e.g. `REPL.complete_nextargs(::typeof(groupby), firstargs::Tuple{AbstractDataFrame}) = names(firstargs[1])`.

Of course that wouldn't work if several packages want to define completions for the same type of the first argument. But at least for types that are owned by the package it would be OK (that's the most common case I guess).

**Comment by bkamins** [comment]
Sure - a nice idea. This would still require also changes to DataFrames.jl (after it is implemented in REPL/IDE) and that is why I have left reopened the issue here.

**Comment by nalimilan** [comment]
OK, filed as https://github.com/JuliaLang/julia/issues/33400.
